domain/removal/uuid.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package removal
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/utils/v4"
+
+	"github.com/juju/juju/internal/uuid"
+)
+
+// UUID represents a upgrade unique identifier.
+type UUID string
+
+// NewUUID returns a new UUID.
+func NewUUID() (UUID, error) {
+	id, err := uuid.NewUUID()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	return UUID(id.String()), nil
+}
+
+// MustNewUUID returns a new UUID or panics.
+func MustNewUUID() UUID {
+	id, err := NewUUID()
+	if err != nil {
+		panic(err)
+	}
+	return id
+}
```

Comment:
This is incorrect. We're not following this pattern anymore. See: https://github.com/juju/juju/blob/main/core/application/testing/testing.go as an example.

That way, the panic doesn't land in production code.

Reply:
Done. Also removed it from where I cribbed this code - `upgrade`.